src/IECoreScene/MeshPrimitive.cpp

```diff
@@ -560,13 +560,37 @@ MeshPrimitivePtr MeshPrimitive::createBox( const Box3f &b )
 	p.push_back( V3f( b.min.x, b.min.y, b.max.z ) );	// 6
 	p.push_back( V3f( b.min.x, b.max.y, b.max.z ) );	// 7
 
-	verticesPerFaceVec.resize( sizeof( verticesPerFace ) / sizeof( int ) );
```

Comment:
It looks like we no longer use `verticesPerFaceVec` or `vertexIdsVec`, but they're still declared up top? If you're interested in simplifying this further, it would seem that with C++11 we can avoid the intermediate vector and just do `new IntVectorData( { 4, 4, 4, 4, 4, 4 } )`.